### PR TITLE
Annotate EC task to align with RHDH requirement

### PR DESCRIPTION
**Pull request by karthikjeeyar**
Fixes:
https://issues.redhat.com/browse/RHTAPBUGS-1179

Description:

Add required annotations to verify-enterprise-contract task so RHDH shows the EC report in the output tab.

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
cc: @jduimovich

File: task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml [hunk 1]
```diff
@@ -103,6 +107,15 @@ spec:
         value: "$(params.HOMEDIR)"
 
   steps:
+    - name: annotate-task
```
Comment (zregvart) [inline task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml:110]: Why is this required? The annotations from the definition are present on the TaskRun. We do not want to allow TaskRuns to modify their own metadata this could lead to all sort of security issues.

**Review by zregvart — CHANGES_REQUESTED** [review]
The task in here is a copy of https://github.com/enterprise-contract/ec-cli/blob/main/tasks/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml and any changes should be made there first.

File: task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml (hunk 1 above)
Comment (karthikjeeyar) [reply task/verify-enterprise-contract/0.1/verify-enterprise-contract.yaml:110]: PAC is removing these custom annotations which is required for RHDH UI to identify the ec task to show the reports, so this seems like a workaround to overcome this issue. Similar fix is done in ACS and SBOM tasks ref- https://github.com/redhat-appstudio/build-definitions/pull/859 . Tagging @jduimovich who can add more details to this.

**Comment by karthikjeeyar** [comment]
@zregvart After talking to John, I realised we dont want to put the RHTAP specific annotations on here and in enterprise-contract repo and moreover this is a temporary workaround until PAC fixes this issue. so closing this PR in favor of https://github.com/redhat-appstudio/tssc-sample-pipelines/pull/32
